### DISCUSSION TITLE
PngBitDepth.Bit8 not work, image still bit32

### DISCUSSION THREAD

**Discussion by df257** [SixLabors/ImageSharp#1480] [Q&A] [answered]
This is my code:

```c#
using(SixLabors.ImageSharp.Image image = SixLabors.ImageSharp.Image.Load(@"E:\c#\WindowsFormsApp1\bin\Debug\test.png")) {
    SixLabors.ImageSharp.Formats.Png.PngEncoder pngEncoder = new SixLabors.ImageSharp.Formats.Png.PngEncoder();
    pngEncoder.CompressionLevel = SixLabors.ImageSharp.Formats.Png.PngCompressionLevel.Level5;
    pngEncoder.BitDepth = SixLabors.ImageSharp.Formats.Png.PngBitDepth.Bit8;
    image.SaveAsPng("test2.png", pngEncoder);
}
```

**Comment by JimBobSquarePants** [comment, marked as answer]
[`PngBitDepth`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.Png.PngBitDepth.html) is calculated at bits per sample or per palette index (not per pixel). This is inline with the specification. To set your colorspace you also need to set the [`PngColorType`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.Png.PngColorType.html) property.

This issue would have been better suited as a QA discussion and as such I've transferred it.
